frontend/src/store/actions.js

```diff
@@ -144,31 +145,38 @@ const handleResponseError = error => {
   const serverError = !maintenanceMode && error.response && error.response.status >= 500
   // Report request timeouts
   const requestTimeout = error.code === 'ECONNABORTED'
+  const unAuthenticated = invalidToken(error.response)
   if (maintenanceMode) {
     location.reload()
   } else if (serverError || requestTimeout) {
     raven.captureException(error)
+  } else if (unAuthenticated) {
+    store.dispatch(setUserLoggedIn(false))
   }
   return Promise.reject(error)
 }
 
 http.interceptors.response.use(
-  response => response,
-  error => handleResponseError(error))
-
-anon.interceptors.response.use(
-  response => response,
+  response => {
+    store.dispatch(setUserLoggedIn(true))
```

Comment:
we still want the anonymous http call, so that every successful http request doesn't mark the user as logged in.